main/celery.py

```diff
@@ -49,241 +32,76 @@
+        input_file = create_tempfile(dfile, ".vcf.gz")
+        input_vcf = vcf.Reader(filename=input_file.name, compressed=True)
+    elif base_name.endswith('.vcf.bz2'):
+        input_file = create_tempfile(dfile, ".vcf.bz2")
+        vcf_file = io.TextIOWrapper(bz2.BZ2File(input_file.name))
+        input_vcf = vcf.Reader(vcf_file)
+    elif base_name.endswith('.vcf'):
+        input_file = create_tempfile(dfile, ".vcf")
+        input_vcf = vcf.Reader(filename=input_file.name)
     else:
-        logger.warn('23andMe header did not conform to expected format.')
+        raise ValueError("Input filename doesn't match .vcf, .vcf.gz, "
+                         'or .vcf.bz2')
 
-    bad_format = False
+    # Check that it can advance one record without error.
+    next(input_vcf)
 
-    while next_line:
-        regex = re.compile(r"""(rs|i)[0-9]+
-            \t[1-9XYM][0-9T]?
-            \t[0-9]+
-            \t[ACGT\-ID][ACGT\-ID]?
-            """, re.VERBOSE)
-        if re.match(regex, next_line):
-            output.write(next_line)
-        else:
-            # Only report this type of format issue once.
-            if not bad_format:
-                bad_format = True
-                logger.warn('23andMe body did not conform to expected format.')
-                logger.warn('Bad format: "%s"', next_line)
-
-        try:
-            next_line = input_file.readline()
-        except StopIteration:
-            next_line = None
-
-    if bad_format:
-        logger.warn('23andMe body did not conform to expected format.')
-
-    return output
+    return input_vcf.metadata
 
 
 def process_file(dfile, access_token, member, metadata):
     try:
-        infile_suffix = dfile['basename'].split(".")[-1]
-        tf_in = tempfile.NamedTemporaryFile(suffix="."+infile_suffix)
-        tf_in.write(requests.get(dfile['download_url']).content)
-        tf_in.flush()
+        vcf_metadata = verify_vcf(dfile)
         tmp_directory = tempfile.mkdtemp()
-        filename_base = '23andMe-genotyping'
-
-        raw_23andme = clean_raw_23andme(tf_in)
-        raw_23andme.seek(0)
-        vcf_23andme = vcf_from_raw_23andme(raw_23andme)
+        base_filename = dfile['basename']
 
         # Save raw 23andMe genotyping to temp file.
-        raw_filename = filename_base + '.txt'
-        raw_filename = temp_join(tmp_directory, raw_filename)
+        if base_filename.endswith('.gz'):
+            base_filename = base_filename[0:-3]
+        elif base_filename.endswith('.bz2'):
+            base_filename = base_filename[0:-4]
+        meta_filename = base_filename + '.metadata.json'
+        raw_filename = temp_join(tmp_directory, meta_filename)
         metadata = {
                     'description':
-                    '23andMe full genotyping data, original format',
-                    'tags': ['23andMe', 'genotyping'],
-                    'creation_date': arrow.get().format(),
-            }
+                    'VCF file metadata',
```

Comment:
For consistency with the current legacy source, I think the description would be `"User-contributed VCF data"` - see https://www.openhumans.org/api/public-data/?source=vcf_data